### PR TITLE
upgrade real-world to net9

### PR DESCRIPTION
**Pull request by kunalspathak**

### CODE REVIEW

**Review by LoopedBard3 — DISMISSED** [review]
LGTM if the test run failures match main or there are no failures.

File: src/benchmarks/real-world/Roslyn/CompilerBenchmarks.csproj [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>$(PERFLAB_TARGET_FRAMEWORKS)</TargetFrameworks>
-    <TargetFrameworks Condition="'$(TargetFrameworks)' == ''">net8.0</TargetFrameworks>
+    <TargetFrameworks>net7.0</TargetFrameworks>
```
Comment (cincuranet) [inline src/benchmarks/real-world/Roslyn/CompilerBenchmarks.csproj:7]: Is this change intentional?

File: src/benchmarks/real-world/ILLink/SampleProject/HelloWorld.template.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 	<PropertyGroup>
 		<OutputType>exe</OutputType>
-		<TargetFramework>net8.0</TargetFramework>
+		<TargetFramework>net9.0</TargetFramework>
```
Comment (cincuranet) [inline src/benchmarks/real-world/ILLink/SampleProject/HelloWorld.template.csproj:4]: I think this should be changed to use `$(PERFLAB_TARGET_FRAMEWORKS)` and the `Condition`. For consistency sake.

File: src/benchmarks/real-world/Roslyn/CompilerBenchmarks.csproj (hunk 1 above)
Comment (kunalspathak) [reply src/benchmarks/real-world/Roslyn/CompilerBenchmarks.csproj]: it was not. fixed it.

File: src/benchmarks/real-world/ILLink/SampleProject/HelloWorld.template.csproj (hunk 2 above)
Comment (kunalspathak) [reply src/benchmarks/real-world/ILLink/SampleProject/HelloWorld.template.csproj]: sure

**Review by cincuranet — DISMISSED** [review]

**Comment by cincuranet** [comment]
@kunalspathak Do you want me to merge it?

**Comment by kunalspathak** [comment]
> @kunalspathak Do you want me to merge it?

sure, once CI is green.

**Comment by LoopedBard3** [comment]
> Do we want to hold off the PR or merge and investigate then? I'll check eurotomorrow either way.

I think we wait until we have had a chance to look into it a little bit, making a decision to merge with the break or not then.

**Comment by kunalspathak** [comment]
> illink is failing with:

It was failing because of the change suggested in https://github.com/dotnet/performance/pull/3511#discussion_r1406591526. I reverted it and things work locally.

**Comment by LoopedBard3** [comment]
Sorry, wrong GitHub button.

**Comment by LoopedBard3** [comment]
The change seems to be caused by the change from TargetFramework to TargetFrameworks in the Illink Sample csproj here: https://github.com/dotnet/performance/pull/3511/files#diff-359fe5c6a7aea7d51cba906e3885f24452af66155f3d0568a4b7d13297278b68L4-R5. 

Something like this update to the publish args in the Utilities.cs -> PublishSampleProject (https://github.com/dotnet/performance/blob/main/src/benchmarks/real-world/ILLink/Utilities.cs#L64) fixed the test locally for me.

``` 
-f {Environment.GetEnvironmentVariable("PERFLAB_TARGET_FRAMEWORKS") ?? "net9.0"}
```

**Review by LoopedBard3 — APPROVED** [review]

**Comment by kunalspathak** [comment]
Are the CI failures known and can we merge the PR?

**Comment by LoopedBard3** [comment]
I will go ahead and make another PR once this one goes in 👍. @cincuranet I think we are good to merge this.
